packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/property-table.tsx

```diff
@@ -0,0 +1,321 @@
+import {
+  DataEditor,
+  GridCell,
+  GridCellKind,
+  GridColumn,
+  Item,
+  DrawCustomCellCallback,
+  DrawHeaderCallback,
+  Theme,
+  EditableGridCell,
+} from "@glideapps/glide-data-grid";
+import { useCallback, useMemo } from "react";
+import "@glideapps/glide-data-grid/dist/index.css";
+import { useTheme } from "@mui/material";
+import { PropertyType } from "@blockprotocol/type-system-web";
+import { pick, capitalize } from "lodash";
+import { EntityResponse } from "../../../../components/hooks/blockProtocolFunctions/knowledge/knowledge-shim";
+import { useBlockProtocolUpdateEntity } from "../../../../components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolUpdateEntity";
+import { useEntityEditor } from "./entity-editor-context";
+
+type Row = {
+  title: string;
+  value: any;
+  dataTypes: string[];
+  propertyTypeId: string;
+};
+
+const columns: GridColumn[] = [
+  {
+    title: "Property",
+    id: "title",
+    width: 250,
+  },
+  {
+    title: "Value",
+    id: "value",
+    grow: 1,
+    width: 300,
+  },
+  {
+    title: "Data type",
+    id: "type",
+    width: 200,
+  },
+];
+
+const indexes: Exclude<keyof Row, "propertyTypeId">[] = [
+  "title",
+  "value",
+  "dataTypes",
+];
+
+const firstColumnPadding = 36;
+const columnPadding = 22;
+
+interface PropertyTableProps {
+  showSearch?: boolean;
+  onSearchClose?: () => void;
+  entity: EntityResponse;
+}
+
+type EnrichedPropertyType = PropertyType & {
+  value: any;
+  propertyTypeId: string;
+  dataTypes: string[];
+};
+
+const getDataTypesOfPropertyType = (
+  propertyType: PropertyType,
+  entity: EntityResponse,
+) => {
+  /** @todo check why propertyValue does not have with a proper type  */
+  return propertyType.oneOf.map((propertyValue: any) => {
+    if (propertyValue?.$ref) {
+      const dataTypeId = propertyValue?.$ref;
+      return (
+        entity.entityTypeRootedSubgraph.referencedDataTypes.find(
+          (val) => val.dataTypeId === dataTypeId,
+        )?.dataType.title ?? "undefined"
+      );
+    }
+
+    return capitalize(propertyValue.type);
+  });
+};
+
+const extractEnrichedPropertyTypesFromEntity = (
+  entity: EntityResponse,
+): EnrichedPropertyType[] => {
+  return Object.keys(entity.properties).map((propertyTypeId) => {
+    const { propertyType } =
+      entity.entityTypeRootedSubgraph.referencedPropertyTypes.find((val) =>
+        val.propertyTypeId.startsWith(propertyTypeId),
+      ) ?? {};
+
+    if (!propertyType) {
+      throw new Error();
+    }
+
+    const dataTypes = getDataTypesOfPropertyType(propertyType, entity);
+
+    return {
+      ...propertyType,
+      value: entity.properties[propertyTypeId],
+      propertyTypeId,
+      dataTypes,
+    };
+  });
+};
+
+export const PropertyTable = ({
+  showSearch,
+  onSearchClose,
+  entity,
+}: PropertyTableProps) => {
+  const { palette } = useTheme();
```

Comment:
This is application-specific data – we can figure out replacing this later, since we haven't settled on a 'how blocks receive styling/theme data' solution.

Reply:
I'll add a comment line explaining this above useTheme. Not adding right now, because I changed where this line exists on the [following PR - WIP](https://github.com/hashintel/hash/pull/1207)